src/diagram/components/ui/expandable-input.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { ChangeEvent, FocusEvent, MouseEvent, useState } from "react";
+import classNames from "classnames";
+import { NumberInput } from "./number-input";
+import { isValidNumber } from "../../utils/validate";
+import { IconExpand } from "../icon-expand";
+
+import "./expandable-input.scss";
+
+interface IProps {
+  error?: boolean;
+  inputType: "text" | "number";
+  lengthToExpand: number;
+  maxLength?: number;
+  placeholder?: string;
+  title: string;
+  value?: string | number;
+  handleBlur: () => void;
+  handleChange?: (evt: ChangeEvent<HTMLTextAreaElement>) => void;
+  handleFocus: (evt: FocusEvent<HTMLInputElement|HTMLTextAreaElement>|MouseEvent<HTMLInputElement|HTMLTextAreaElement>) => void;
+  handleKeyDown?: (evt: React.KeyboardEvent) => void;
+  handleToggle: () => void;
+  setRealValue: (value: number | undefined) => void;
```

Comment:
Since text inputs won't want this, you could make it optional and only render a `NumberInput` when `inputType === "number" && setRealValue`. In other words, if someone tries to render an `ExpandableInput` with `inputType === "number"` but without passing in a `setRealValue`, you can just render a `textarea` instead. If you go this route, I think it would make sense to add a comment explaining that `setRealValue` is required to render an expandable `NumberInput`.